db/fetcher/fetcher.go

```diff
@@ -27,6 +27,23 @@ import (
 	"github.com/sourcenetwork/defradb/request/graphql/parser"
 )
 
+type Stats struct {
```

Comment:
suggestion: Perhaps more consistent naming, I wanted to avoid the use of the word `metrics` specifically as metrics are a bit more higher level and now days mean something specific i.e usage with instrumentation of histograms, gauge and etc. For consistency can name this `ExecInfo`. Or if you want to keep the word Stats (I don't mind this word) then perhaps `ExecStats` or `ExecutionStats` and rename all other instances where we use the `ExecInfo` terminology.